src/PFXImportPowershell/EncryptionUtilities/Source/ManagedRSAEncryption.cs

```diff
@@ -157,14 +197,20 @@ public bool TryGenerateLocalRSAKey(string providerName, string keyName, int keyL
         public void DestroyLocalRSAKey(string providerName, string keyName)
         {
             CngProvider provider = new CngProvider(providerName);
+            CngKeyOpenOptions cngOp = CngKeyOpenOptions.MachineKey;
 
-            if (CngKey.Exists(keyName, provider))
+            if (!CngKey.Exists(keyName, provider))
             {
-                using (CngKey key = CngKey.Open(keyName, provider))
+                cngOp = CngKeyOpenOptions.UserKey;
+                if (!CngKey.Exists(keyName, provider, cngOp))
                 {
-                    key.Delete();
+                    return;
                 }
             }
+            using (CngKey key = CngKey.Open(keyName, provider, cngOp))
+            {
+                key.Delete();
+            }
```

Comment:
A `using` disposes in the end. Would deleting the key while using `using` cause issues? Or, will the implicit `finally` take care of it?

Reply:
I think we are fine here. "Delete" in c# doesn't unallocated an object in C#. Clean up of resources is handled in the "Dispose" function that is called in the finally.